src/app/meme-dialog/meme-dialog.component.ts

```diff
@@ -0,0 +1,46 @@
+import {Component, EventEmitter, Input, OnInit, Output} from '@angular/core';
+import {MatDialogModule} from '@angular/material';
+import {Utils} from '../utils';
+
+@Component({
+  selector: 'app-meme-dialog',
+  templateUrl: './meme-dialog.component.html',
+  providers: [MatDialogModule],
+  styleUrls: ['./meme-dialog.component.scss']
+})
+export class MemeDialogComponent implements OnInit {
+  @Input() username: string;
+  @Input() image: string;
+  @Input() totalVote: number;
+  @Input() myVote: number;
+
+  @Output() notifyCard: EventEmitter<number> = new EventEmitter<number>();
+
+  constructor() {
+  }
+
+  ngOnInit() {
+  }
+
+  maxCardWidth(): number {
+    return Utils.screenWidth * 0.70;
+  }
+
+  minCardWidth(): number {
+    return Utils.screenWidth * 0.5;
+  }
+
+  maxDialogHeight(): number {
+    return Utils.screenHeight * 0.9;
+  }
+
+  // I'd rather do these in a more natural way, e.g. property binding but I can't seem to get it to work properly
+  onClickUpVote() {
```

Comment:
You should talk to the meme service instead. Do a similar thing to how the meme card operates and if the meme card wants to update, it can do it itself after the dialog closes.

Reply:
That would look weird from the user's perspective though, if they can see the up/downvotes of the card behind the dialog.

Reply:
Are you suggesting the background page should appear to freeze while the dialog is open, then update when the dialog closes? I just don't really understand how duplicating the method like that is better but I'm not super familiar with best practices